tests/app/main/views/test_platform_admin.py

```
@@ -1079,6 +1079,54 @@ def test_get_notifications_sent_by_service_validates_form(mocker, client_request
     mock_get_stats_from_api.assert_not_called()
 
 
+def test_usage_for_all_services_when_no_results_for_date(client_request, platform_admin_user, mocker):
+    client_request.login(platform_admin_user)
+
+    mocker.patch("app.main.views.platform_admin.billing_api_client.get_usage_for_all_services",
+                 return_value=[])
+
+    page = client_request.post('main.usage_for_all_services',
+                               _expected_status=200,
+                               _data={'start_date': '2019-01-01', 'end_date': '2019-03-31'})
+
+    errors = page.select('.error-message')
```

you can use `error = page.select_one('.error-message')` to avoid having to loop through a one-long array

---

also this was hiding an error - errors was empty, so the loop below wasn't being entered and no asserts were given